File: androidApp/src/main/kotlin/com/chaoticware/discrobble/android/security/StoredIntegrationTokenSet.kt
package com.chaoticware.discrobble.android.security

import com.chaoticware.discrobble.android.auth.AuthProvider

data class StoredIntegrationTokenSet(
    val provider: AuthProvider,
    val username: String,
    val accessToken: String,
    val accessSecret: String?,
    val issuedAt: String,
    val expiresAt: String?,
) {
    override fun toString(): String {
        val redactedAccessSecret = if (accessSecret == null) "null" else "[REDACTED]"
        return "StoredIntegrationTokenSet(provider=$provider, username=$username, accessToken=[REDACTED], accessSecret=$redactedAccessSecret, issuedAt=$issuedAt, expiresAt=$expiresAt)"
    }
}

File: iosApp/Discrobble/Security/StoredIntegrationTokenSet.swift
import Foundation

struct StoredIntegrationTokenSet: Codable, Equatable {
    let provider: AuthProvider
    let username: String
    let accessToken: String
    let accessSecret: String?
    let issuedAt: Date
    let expiresAt: Date?

    enum CodingKeys: String, CodingKey {
        case provider
        case username
        case accessToken = "access_token"
        case accessSecret = "access_secret"
        case issuedAt = "issued_at"
        case expiresAt = "expires_at"
    }
}
